python/lsst/ts/cbp/csc.py

```diff
@@ -120,6 +120,11 @@ async def telemetry(self):
         """Publish the updated telemetry."""
         self.log.debug("Begin sending telemetry")
         while True:
+            if not self.component.connected and self.component.should_be_connected:
+                await self.fault(
+                    ErrorCode.CONNECTION_FAILED, report="Lost connection to controller."
+                )
+                return
```

Comment:
And you have this in another file:

```python
                    except ConnectionError:
                        self.log.exception("Lost connection.")
                        await self.csc.fault(code=1, report="Lost Connection")
```

Maybe use `ErrorCode.CONNECTION_FAILED` there?
But you can see you have two different classes are checking the similar thing and do the similar action.
I would suggest to have another ticket to decouple them and put the checking or action to the CSC only.